packages/spezi-firebase-fhir/src/resources/coverage.ts

```diff
@@ -0,0 +1,93 @@
+//
+// This source file is part of the Stanford Biodesign Digital Health Spezi Firebase Remote Notifications open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import { FhirDomainResource } from './domainResourceClass.js'
+import { type Coverage } from 'fhir/r4b.js'
+import { z, type ZodType } from 'zod'
+import { domainResourceSchema } from '../elements/domainResource.js'
+import {
+  identifierSchema,
+  backboneElementSchema,
+  elementSchema,
+  codeableConceptSchema,
+  periodSchema,
+  referenceSchema,
+  quantitySchema,
+  moneySchema,
+  positiveIntSchema,
+  stringSchema,
+  booleanSchema,
+} from '../elements/index.js'
+
+const coverageStatusSchema = z.enum([
+  'active',
+  'cancelled',
+  'draft',
+  'entered-in-error',
+])
+
+export const untypedCoverageSchema = z.lazy(() =>
+  domainResourceSchema.extend({
+    resourceType: z.literal('Coverage').readonly(),
+    identifier: identifierSchema.array().optional(),
+    status: coverageStatusSchema,
+    _status: elementSchema.optional(),
+    type: codeableConceptSchema.optional(),
+    policyHolder: referenceSchema.optional(),
+    subscriber: referenceSchema.optional(),
+    subscriberId: stringSchema.optional(),
+    _subscriberId: elementSchema.optional(),
+    beneficiary: referenceSchema,
+    dependent: stringSchema.optional(),
+    _dependent: elementSchema.optional(),
+    relationship: codeableConceptSchema.optional(),
+    period: periodSchema.optional(),
+    class: backboneElementSchema
+      .extend({
+        type: codeableConceptSchema,
+        _type: elementSchema.optional(),
```

Comment:
_type is not needed here, since type is a CodeableConcept and not a string or another primitive type.